src/assets/scripts/client/airport/MapCollection.js

```diff
@@ -0,0 +1,140 @@
+import _isArray from 'lodash/isArray';
+import _isObject from 'lodash/isObject';
+import BaseCollection from '../base/BaseCollection';
+import MapModel from './MapModel';
+
+/**
+ * Collection of `MapModel` objects
+ *
+ * Provides methods to create `MapModel` objects, used by `AirportModel`
+ * and to extract video map line data to be rendered by the
+ * `CanvasController`
+ *
+ * @class MapCollection
+ * @extends BaseCollection
+ */
+export default class MapCollection extends BaseCollection {
+    /**
+     * @constructor
+     * @param mapJson {object}
+     * @param airportPositionModel {StaticPositionModel}
+     * @param magneticNorth {number}
+     */
+    constructor(mapJson, airportPositionModel, magneticNorth) {
+        super();
+
+        if (!_isArray(mapJson) && !_isObject(mapJson)) {
+            throw new TypeError(
+                `Invalid parameter passed to MapCollection. Expected an array or object but found ${typeof mapJson}`
+            );
+        }
+
+        /**
+         * @inherited
+         * @memberof BaseCollection
+         * @property _items
+         * @type {array<MapModel>}
+         * @default []
+         */
+
+        /**
+         * @inherited
+         * @memberof BaseCollection
+         * @property length
+         * @type {number}
+         * @default #_items.length
+         */
+
+        /**
+         * @property _airportPositionModel
+         * @type {StaticPositionModel}
+         * @default null
+         * @private
+         */
+        this._airportPositionModel = null;
+
+        /**
+         * @property _magneticNorth
+         * @type {number}
+         * @default null
+         * @private
+         */
+        this._magneticNorth = null;
+
+        this._init(mapJson, airportPositionModel, magneticNorth);
+    }
+
+    /**
+     * Public fascade for `#_items`
+     *
+     * @property maps
+     * @return {array<MapModel>}
+     */
+    get maps() {
+        return this._items;
+    }
+
+    /**
+     * @for MapCollection
+     * @method getVisibleMapLines
+     * @return {array<array<object>>}
+     */
+    getVisibleMapLines() {
+        const filtered = this._items.filter((map) => {
+            return !map.isHidden && map.lines.length
+        });
+        
+        return filtered.map((map) => {
+            return map.lines;
+        });
+    }
+
+    /**
+     * Initialize the instance
+     *
+     * @for MapCollection
+     * @method _init
+     * @param mapJson {object}
+     * @param airportPositionModel {StaticPositionModel}
+     * @param magneticNorth {number}
+     */
+    _init(mapJson, airportPositionModel, magneticNorth) {
+        this._airportPositionModel = airportPositionModel;
+        this._magneticNorth = magneticNorth;
+
+        this._buildMapModels(mapJson);
+    }
+
+    /**
+     * Tear down the instance and destroy any instance property values
+     *
+     * @for MapCollection
+     * @method destroy
+     */
+    destroy() {
```

Comment:
you'll want to reset all class properties in this method

Reply:
Now with the airportPositionModel and magneticNorth variables moved to being params, this is essentially empty.

Or should this include _items property too? I took the usage from RunwayCollection which doesn't destroy _items.